libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountBuilder.kt

```diff
@@ -0,0 +1,529 @@
+    private var communityUrl: String? = null
+    private var firstName: String? = null
+    private var lastName: String? = null
+    private var displayName: String? = null
+    private var email: String? = null
+    private var photoUrl: String? = null
+    private var thumbnailUrl: String? = null
+    private var lightningDomain: String? = null
+    private var lightningSid: String? = null
+    private var vfDomain: String? = null
+    private var vfSid: String? = null
+    private var contentDomain: String? = null
+    private var contentSid: String? = null
+    private var csrfToken: String? = null
+    private var nativeLogin = false
+    private var language: String? = null
+    private var locale: String? = null
+    private var cookieClientSrc: String? = null
+    private var cookieSidClient: String? = null
+    private var sidCookieName: String? = null
+    private var clientId: String? = null
+    private var additionalOauthValues: Map<String, String>? = null
+    private var allowUnset = true
+
+    /**
+     * Set fields from token end point response
+     * @param tr token endpoint response
+     * @return Instance of this class
+     */
+    fun populateFromTokenEndpointResponse(tr: TokenEndpointResponse?): UserAccountBuilder {
+        if (tr == null) {
+            return this
+        }
+
+        return authToken(tr.authToken)
+            .refreshToken(tr.refreshToken)
+            .instanceServer(tr.instanceUrl)
+            .idUrl(tr.idUrl)
+            .orgId(tr.orgId)
+            .userId(tr.userId)
+            .communityId(tr.communityId)
+            .communityUrl(tr.communityUrl)
+            .additionalOauthValues(tr.additionalOauthValues)
+            .lightningDomain(tr.lightningDomain)
+            .lightningSid(tr.lightningSid)
+            .vfDomain(tr.vfDomain)
+            .vfSid(tr.vfSid)
+            .contentDomain(tr.contentDomain)
+            .contentSid(tr.contentSid)
+            .csrfToken(tr.csrfToken)
+            .cookieClientSrc(tr.cookieClientSrc)
+            .cookieSidClient(tr.cookieSidClient)
+            .sidCookieName(tr.sidCookieName)
+    }
+
+    /**
+     * Set fields from id service response
+     * @param id id service response
+     * @return Instance of this class
+     */
+    fun populateFromIdServiceResponse(id: IdServiceResponse?): UserAccountBuilder {
+        if (id == null) {
+            return this
+        }
+
+        return username(id.username)
+            .firstName(id.firstName)
+            .lastName(id.lastName)
+            .displayName(id.displayName)
+            .email(id.email)
+            .photoUrl(id.pictureUrl)
+            .thumbnailUrl(id.thumbnailUrl)
+            .language(id.language)
+            .locale(id.locale)
+    }
+
+    /**
+     * Set fields from an existing UserAccount
+     * @param userAccount user account.
+     * @return Instance of this class
+     */
+    fun populateFromUserAccount(userAccount: UserAccount): UserAccountBuilder {
```

Comment:
Used during refresh to apply new values onto existing objects (to use populateXXX, we need a UserAccountBuilder with the values of an existing UserAccount).